lib/stove-testing-e2e-elasticsearch/src/main/kotlin/com/trendyol/stove/testing/e2e/elasticsearch/ElasticsearchSystem.kt

```diff
@@ -0,0 +1,181 @@
+@file:Suppress("UNCHECKED_CAST")
+
+package com.trendyol.stove.testing.e2e.elasticsearch
+
+import arrow.core.getOrElse
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.trendyol.stove.testing.e2e.containers.withProvidedRegistry
+import com.trendyol.stove.testing.e2e.database.DatabaseSystem
+import com.trendyol.stove.testing.e2e.database.DocumentDatabaseSystem
+import com.trendyol.stove.testing.e2e.elasticsearch.ElasticsearchSystem.Constants.ONE_MINUTE
+import com.trendyol.stove.testing.e2e.elasticsearch.ElasticsearchSystem.Constants.ONE_SECOND
+import com.trendyol.stove.testing.e2e.serialization.StoveJacksonJsonSerializer
+import com.trendyol.stove.testing.e2e.serialization.StoveJsonSerializer
+import com.trendyol.stove.testing.e2e.system.TestSystem
+import com.trendyol.stove.testing.e2e.system.abstractions.ExposesConfiguration
+import com.trendyol.stove.testing.e2e.system.abstractions.RunAware
+import com.trendyol.stove.testing.e2e.system.abstractions.SystemNotRegisteredException
+import org.apache.http.HttpHost
+import org.apache.http.client.config.RequestConfig.Builder
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder
+import org.elasticsearch.action.delete.DeleteRequest
+import org.elasticsearch.action.get.GetRequest
+import org.elasticsearch.action.index.IndexRequest
+import org.elasticsearch.client.Request
+import org.elasticsearch.client.RequestOptions
+import org.elasticsearch.client.RestClient
+import org.elasticsearch.client.RestClientBuilder
+import org.elasticsearch.client.RestHighLevelClient
+import org.elasticsearch.common.xcontent.XContentType
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.testcontainers.elasticsearch.ElasticsearchContainer
+import kotlin.reflect.KClass
+
+data class ElasticsearchSystemOptions(
+    val configureExposedConfiguration: (ElasticSearchExposedConfiguration) -> List<String> = { _ -> listOf() },
+    val jsonSerializer: StoveJsonSerializer = StoveJacksonJsonSerializer(),
+)
+
+fun TestSystem.withElasticsearch(
+    index: String,
+    registry: String = "docker.elastic.co/",
```

Comment:
It would be better to get these values into `ElasticsearchSystemOptions` with:
- DefaultIndex
- `ContainerOptions(registry, imageVersion)`

Reply:
Actually there is no any default index for elasticsearch

Reply:
Changed according to suggestion, but without default index property